From 9786ba8f6daee26acc79f64a056d03a464214d27 Mon Sep 17 00:00:00 2001
From: Marian Chereji <marian.chereji@freescale.com>
Date: Thu, 17 Oct 2013 11:33:57 +0300
Subject: [PATCH 291/429] dpa_offload: Fix IPv4 options size field
 inconsistency

Due to a bad sync the ipv4_header defines the attribute options_size as
uint8_t, while the compat_ipv4_header defines it as unsigned int. The
compat_ipv4_header must define this attribute as uint8_t as well to
avoid problems when casting from one structure to the other.

Signed-off-by: Marian Chereji <marian.chereji@freescale.com>
Change-Id: I98df663fda9d0ea19f0e17ac71f99512b06c5bb6
Reviewed-on: http://git.am.freescale.net:8181/5742
Tested-by: Review Code-CDREVIEW <CDREVIEW@freescale.com>
Reviewed-by: Richard Schmitt <B43082@freescale.com>
[origin patch is from QorIQ-SDK-V1.5-20131219-yocto]
Signed-off-by: Yang Wei <Wei.Yang@windriver.com>
---
 .../staging/fsl_dpa_offload/dpa_classifier_ioctl.h |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/staging/fsl_dpa_offload/dpa_classifier_ioctl.h b/drivers/staging/fsl_dpa_offload/dpa_classifier_ioctl.h
index df0fee2..88a157c 100644
--- a/drivers/staging/fsl_dpa_offload/dpa_classifier_ioctl.h
+++ b/drivers/staging/fsl_dpa_offload/dpa_classifier_ioctl.h
@@ -364,7 +364,7 @@ struct compat_ioc_dpa_cls_hm_vlan_params {
 struct compat_ipv4_header {
 	struct iphdr			header;
 	compat_uptr_t			options;
-	unsigned int			options_size;
+	uint8_t				options_size;
 };
 
 struct dpa_cls_compat_hm_nat_pt_params {
-- 
1.7.5.4

